client/signup/steps/site-type/form.jsx

```diff
@@ -44,29 +55,86 @@ class SiteTypeForm extends Component {
 		this.props.submitForm( type );
 	};
 
-	renderSegmentOptions() {
-		return getAllSiteTypes().map( siteTypeProperties => (
-			<Card
-				className="site-type__option"
-				key={ siteTypeProperties.id }
-				displayAsLink
-				data-e2e-title={ siteTypeProperties.slug }
-				tagName="button"
-				onClick={ this.handleSubmit.bind( this, siteTypeProperties.slug ) }
-			>
-				<strong className="site-type__option-label">{ siteTypeProperties.label }</strong>
-				<span className="site-type__option-description">{ siteTypeProperties.description }</span>
+	handleSubmitOther = () => {
+		const fieldValue = this.state.otherValue ? 'other—' + this.state.otherValue : 'other–other';
+
+		this.props.recordTracksEvent( 'calypso_signup_actions_submit_site_type', {
```

Comment:
Since the event is fired in `this.handleSubmit()`, this causes it to fire twice.

![Screen Shot 2019-04-10 at 9 54 28 AM](https://user-images.githubusercontent.com/942359/55885908-c58bfe00-5b78-11e9-8c5d-8f46c9b981ff.png)

Also, this is probably a miscommunication on my end, but in order to segment in the funnels, it'd be better to submit a prop of `value: 'other'` and additionally a prop with the other value. Maybe something like this:

```
handleSubmit = type => {
		this.props.recordTracksEvent( 'calypso_signup_actions_submit_site_type', {
			value: type,
			user_input_site_type: this.state.otherValue || null,
		} );

		this.setState( { siteType: type } );

		this.props.submitForm( type );
};

handleSubmitOther = () => this.handleSubmit( 'other' );
```

Reply:
Of course, makes total sense. Thanks for spotting it.
Re saving the input value itself—yes, it seems like I misunderstood. I guess I do not see how the type of detailed data like `user_input_site_type` can be useful stored as props in tracks. That said, happy to make the change.